### PR TITLE
RDM-6387: functional test cases for "Get case ids" endpoint

### PR DESCRIPTION
**Pull request by SivaKanukolluQA**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-6387

### Change description ###
functional test cases for "Get case ids" endpoint

**Does this PR introduce a breaking change?** (check one with "x")
[ ] Yes
[x] No

### CODE REVIEW

File: src/aat/resources/features/F-047/F-047_Grant_Access.td.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "request": {
+    "_extends_": "Common_Request",
+    "pathVariables": {
+      "uid": "[[DYNAMIC]]",
+      "jid": "AUTOTEST1",
+      "ctid": "AAT",
+      "cid": "${[scenarioContext][parentContext][childContexts][Standard_Full_Case_Creation_Data][testData][actualResponse][body][id]}"
+    },
+    "body": {
+      "id": "F047UserIdSiva"
```
Comment (nitishsingla1234) [inline src/aat/resources/features/F-047/F-047_Grant_Access.td.json]: Kindly use generic names only.

File: src/aat/resources/features/F-047/F-047_Test_Data_Base.td.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "_guid_": "F-047_Test_Data_Base",
+  "productName": "CCD Data Store",
+  "operationName": "Get case ids",
+  "method": "GET",
+  "uri": "/caseworkers/{uid}/jurisdictions/{jid}/case-types/{ctid}/cases/ids?userId={userid}",
```
Comment (nitishsingla1234) [inline src/aat/resources/features/F-047/F-047_Test_Data_Base.td.json]: "uri": "/caseworkers/{uid}/jurisdictions/{jid}/case-types/{ctid}/cases/ids?userId={userid}"
Ideally, the parameter "userId={userid}" should be set using QueryParams.

File: src/aat/resources/features/F-047/F-047_Grant_Access.td.json (hunk 1 above)
Comment (SivaKanukolluQA) [reply src/aat/resources/features/F-047/F-047_Grant_Access.td.json]: Updated: F047UserIdTest

File: src/aat/resources/features/F-047/F-047_Test_Data_Base.td.json (hunk 2 above)
Comment (SivaKanukolluQA) [reply src/aat/resources/features/F-047/F-047_Test_Data_Base.td.json]: Updated.
    "queryParams": {
      "userid": "F047UserIdTest"
    }
  "uri": "/caseworkers/{uid}/jurisdictions/{jid}/case-types/{ctid}/cases/ids?userId"

File: src/aat/resources/features/F-047/F-047.feature [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  @S-100
+  Scenario: must return 403 when request provides authentic credentials without authorised access to the operation
+    Given a user with [an active profile in CCD]
+    When a request is prepared with appropriate values
+    And the request [does not provide authorised access to the operation]
+    And it is submitted to call the [Get case ids] operation of [CCD Data Store]
+    Then a negative response is received
+    And the response [contains an HTTP-403 Forbidden]
+    And the response has all other details as expected
+
```
Comment (shrivsksavit) [inline src/aat/resources/features/F-047/F-047.feature]: Believe we are missing some 400 and 404 test could you please have a look

Comment (SivaKanukolluQA) [reply src/aat/resources/features/F-047/F-047.feature]: I feel those are not necessary and implemented scenarios as mentioned in the excel.

Comment (MSancaktutar) [reply src/aat/resources/features/F-047/F-047.feature]: These scenarios should really be added, as they are really apparent tests to be carried out. The Excel list contains place holder lines for this kind of scenarios to be added. The obvious ones like these can be added during the sprint work.

File: src/aat/resources/features/F-047/F-047.feature [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  @S-570 @Ignore # Response code mismatch, expected: 400, actual: 500
+  Scenario: must return negative response HTTP-400 when request contains a malformed user ID
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-047/F-047.feature]: Can you please add a defect number for this issue?

File: src/aat/resources/features/F-047/F-047.feature [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    Given a user with [an inactive profile in CCD]
+    When  a request is prepared with appropriate values
+    And   the request [contains a malformed user ID]
+    And   it is submitted to call the [Get case ids] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-400]
+    And   the response has all other details as expected
+
+  @S-571 @Ignore # Response code mismatch, expected: 400, actual: 500
+  Scenario: must return negative response HTTP-400 when request contains a malformed case type ID
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-047/F-047.feature]: Can you please add defect number?
Also add this scenario in RDM-6387.

File: src/aat/resources/features/F-047/F-047.feature [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    Given a user with [an active profile in CCD]
+    When  a request is prepared with appropriate values
+    And   the request [contains a malformed case type ID]
+    And   it is submitted to call the [Get case ids] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-400]
+    And   the response has all other details as expected
+
+  @S-572 @Ignore # Response code mismatch, expected: 400, actual: 403
+  Scenario: must return negative response HTTP-400 when request contains a malformed jurisdiction ID
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-047/F-047.feature]: Can you please add defect number?
Also add this scenario in RDM-6387.

File: src/aat/resources/features/F-047/S-572.td.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  "_guid_": "S-572",
+  "_extends_": "F-047_Test_Data_Base",
+  "specs": [
+    "an active profile in CCD",
+    "contains a malformed jurisdiction ID",
+    "code is HTTP-400"
+  ],
+  "request": {
+    "pathVariables": {
+      "uid": "dummyAuthentication"
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-047/S-572.td.json]: uid should be correct value.
Please update jid with a malformed value.
Also add this scenario in jira RDM-6387.

File: src/aat/resources/features/F-047/F-047.feature [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    Given a user with [an active profile in CCD]
+    When  a request is prepared with appropriate values
+    And   the request [contains a malformed jurisdiction ID]
+    And   it is submitted to call the [Get case ids] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-400]
+    And   the response has all other details as expected
+
+  @S-573 @Ignore # Response code mismatch, expected: 400, actual: 403
+  Scenario: must return negative response HTTP-400 when request contains a non-existing jurisdiction ID
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-047/F-047.feature]: Please update defect number for this issue.
Also add this scenario in Jira RDM-6387.

File: src/aat/resources/features/F-047/S-573.td.json [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+  "request": {
+    "pathVariables": {
+      "jid": "non-existing jurisdiction ID"
+    }
+  },
+  "expectedResponse": {
+    "_extends_": "Common_400_Response",
+    "body": {
+      "exception": "[[ANY]]",
+      "message": "Missing start trigger token",
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-047/S-573.td.json]: Seems this message is not a suitable one for this non-existing jurisdiction ID scenario.
Please update the message if you know.

File: src/aat/resources/features/F-047/F-047.feature [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+    When  a request is prepared with appropriate values
+    And   the request [contains a non-existing jurisdiction ID]
+    And   it is submitted to call the [Get case ids] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-400]
+    And   the response has all other details as expected
+
+  @S-574 @Ignore #Response code mismatch, expected: 400, actual: 200"
+  Scenario: must return negative response HTTP-404 when request contains a non-existing case type ID
+    Given a user with [an active profile in CCD]
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-047/F-047.feature]: Please update the defect number.
Also please update comment : expected code is 404.
Also add this scenario in Jira RDM-6387

File: src/aat/resources/features/F-047/S-574.td.json [hunk 11]
```diff
@@ -0,0 +1,26 @@
+{
+  "title": "must return 200 and a list of case ids a user has access to",
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-047/S-574.td.json]: Title is wrong . This is a negative scenario. Please update as per feature file.

File: src/aat/resources/features/F-047/F-047.feature [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    When  a request is prepared with appropriate values
+    And   the request [contains a non-existing case type ID]
+    And   it is submitted to call the [Get case ids] operation of [CCD Data Store]
+    Then  a negative response is received
+    And   the response [code is HTTP-404]
+    And   the response has all other details as expected
+
+  @S-575
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-047/F-047.feature]: Please add this scenario in Jira RDM-6387.

File: src/aat/resources/features/F-047/S-575.td.json [hunk 13]
```diff
@@ -0,0 +1,21 @@
+{
+  "title": "must return 200 and a list of case ids a user has access to",
+  "_guid_": "S-575",
```
Comment (SivaKanukolluQA) [inline src/aat/resources/features/F-047/S-575.td.json]: Title is wrong. This is a  negative scenario.  Please update as per feature file.

File: src/aat/resources/features/F-047/F-047.feature (hunk 4 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-047/F-047.feature]: done

File: src/aat/resources/features/F-047/F-047.feature (hunk 5 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-047/F-047.feature]: done

File: src/aat/resources/features/F-047/F-047.feature (hunk 6 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-047/F-047.feature]: done

File: src/aat/resources/features/F-047/F-047.feature (hunk 8 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-047/F-047.feature]: done

File: src/aat/resources/features/F-047/F-047.feature (hunk 10 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-047/F-047.feature]: done

File: src/aat/resources/features/F-047/F-047.feature (hunk 12 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-047/F-047.feature]: done

File: src/aat/resources/features/F-047/S-572.td.json (hunk 7 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-047/S-572.td.json]: done

File: src/aat/resources/features/F-047/S-573.td.json (hunk 9 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-047/S-573.td.json]: done

File: src/aat/resources/features/F-047/S-574.td.json (hunk 11 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-047/S-574.td.json]: done

File: src/aat/resources/features/F-047/S-575.td.json (hunk 13 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-047/S-575.td.json]: done

**Comment by SivaKanukolluQA** [comment]
QA approved.

File: src/aat/resources/features/F-047/F-047.feature (hunk 3 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-047/F-047.feature]: added

**Pull request merged**